imap_processing/swe/l1b/swe_l1b.py

```diff
@@ -1,48 +1,928 @@
+        dims=["spin_sector"],
+        attrs=cdf_attrs.get_variable_attributes(
+            "spin_sector_label", check_schema=False
+        ),
+    )
+
+    cycle = xr.DataArray(
+        np.arange(swe_constants.N_QUARTER_CYCLES),
+        name="cycle",
+        dims=["cycle"],
+        attrs=cdf_attrs.get_variable_attributes("cycle", check_schema=False),
+    )
+
+    cycle_label = xr.DataArray(
+        cycle.values.astype(str),
+        name="cycle_label",
+        dims=["cycle"],
+        attrs=cdf_attrs.get_variable_attributes("cycle_label", check_schema=False),
+    )
+
+    cem_id = xr.DataArray(
+        np.arange(swe_constants.N_CEMS, dtype=np.int8),
+        name="cem_id",
+        dims=["cem_id"],
+        attrs=cdf_attrs.get_variable_attributes("cem_id", check_schema=False),
+    )
+
+    # NOTE: LABL_PTR_3 should be CDF_CHAR.
+    cem_id_label = xr.DataArray(
+        cem_id.values.astype(str),
+        name="cem_id_label",
+        dims=["cem_id"],
+        attrs=cdf_attrs.get_variable_attributes("cem_id_label", check_schema=False),
+    )
+
+    # Add science data and it's associated metadata into dataset.
+    # SCIENCE_DATA has array of this shape:
+    #   (n, 24, 30, 7)
+    #   n = total number of full cycles
+    #   24 rows --> 24 esa voltage measurements
+    #   30 columns --> 30 spin angle measurements
+    #   7 elements --> 7 CEMs counts
+    #
+    # The metadata array will need to have this shape:
+    #   (n, 4)
+    #   n = total number of full cycles
+    #   4 rows --> metadata for each full cycle. Each element of 4 maps to
+    #              metadata of one quarter cycle.
+
+    # Create the dataset
+    dataset = xr.Dataset(
+        coords={
+            "epoch": epoch_time,
+            "esa_step": esa_step,
+            "spin_sector": spin_sector,
+            "cem_id": cem_id,
+            "cycle": cycle,
+            "esa_step_label": esa_step_label,
+            "spin_sector_label": spin_sector_label,
+            "cem_id_label": cem_id_label,
+            "cycle_label": cycle_label,
+        },
+        attrs=cdf_attrs.get_global_attributes("imap_swe_l1b_sci"),
+    )
+
+    dataset["science_data"] = xr.DataArray(
+        count_rate,
+        dims=["epoch", "esa_step", "spin_sector", "cem_id"],
+        attrs=cdf_attrs.get_variable_attributes("science_data"),
+    )
+    dataset["acquisition_time"] = xr.DataArray(
+        acq_time,
+        dims=["epoch", "esa_step", "spin_sector"],
+        attrs=cdf_attrs.get_variable_attributes("acquisition_time"),
+    )
+    dataset["acq_duration"] = xr.DataArray(
+        acq_duration,
+        dims=["epoch", "esa_step", "spin_sector"],
+        attrs=cdf_attrs.get_variable_attributes("acq_duration"),
+    )
+
+    dataset["esa_energy"] = xr.DataArray(
+        esa_energies,
+        dims=["epoch", "esa_step", "spin_sector"],
+        attrs=cdf_attrs.get_variable_attributes("esa_energy"),
+    )
+
+    # create xarray dataset for each metadata field
```

Comment:
Can you change all of the below from `metadata_field` to `data_field`. The metadata is really the text associated with the data.
```suggestion
    # create xarray dataarray for each data field
```